nntrainer/tensor/tensor_wrap_specs.h

```diff
@@ -70,6 +75,75 @@ typedef std::tuple<TensorDim, Tensor::Initializer, bool, const std::string,
                    TensorLifespan>
   VarGradSpec;
 
+/**
+ * @brief Tensor Specification which describes how this tensor should be
+ * allocated and managed
+ *
+ */
+struct TensorSpecV2 {
+
+  /**
+   * @brief Tensor is being managed by nntrainer, this enum defines how the
+   * value should be recognized inside nntrainer tensor managing scheme.
+   *
+   */
+  enum class RequestType {
+    PLACEHOLDER, /**< Placeholder defines that nntrainer should never care about
+                    the memory inside the particualar tensor */
+    UNIQUE, /**< Unique means a simple tensor that will be owned explicitly the
+               current request */
+    READ_ONLY_VIEW, /**< Readonly view defines a view of which ownership of @a
+                       underlying memory is at another tensor, also hinting
+                       nntrainer that the operation upon this particular tensor
+                       will never change value of the underlying memory */
+    MAYBE_MODIFYING_VIEW, /**< Maybe modifying view defines a (possible) view of
```

Comment:
consider the following nomenclature:
VIEW: read-only access to memory with size <= source size
SHARED: read-write access to memory with size <= source size
LINK: (inspired by symlink) link to the original tensor

Reply:
I'll open a issue regarding this.